app/models/miq_filter.rb

```diff
@@ -101,11 +101,14 @@ def self.apply_belongsto_filters(inputs, bfilters)
     return []     if inputs.nil?
     return inputs if bfilters.empty?
 
+    vcmeta_index = bfilters.index_with { |tag| belongsto2object_list(tag) }
+
     filtered = []
     inputs.each do |p|
       bfilters.each do |tag|
+        vcmeta_list = vcmeta_index[tag]
```

Comment:
I do like this, does it just make sense to use a hash with lookup built in instead of the lookup loop up front?